geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/VisibilityIterator.scala

```diff
@@ -0,0 +1,38 @@
+/**********************************************************************
+ * Copyright (c) 2013-2021 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ **********************************************************************/
+
+package org.locationtech.geomesa.accumulo.iterators
+
+import org.apache.accumulo.core.client.IteratorSetting
+import org.apache.accumulo.core.client.admin.TableOperations
+import org.apache.accumulo.core.iterators.IteratorUtil.IteratorScope
+import org.apache.accumulo.core.iterators.user.ReqVisFilter
+import org.locationtech.geomesa.accumulo.data.AccumuloDataStore
+import org.opengis.feature.simple.SimpleFeatureType
+
+object VisibilityIterator {
+
+  val Name = "ReqVisFilter"
+
+  def set(tableOps: TableOperations, table: String): Unit =
+    tableOps.attachIterator(table, new IteratorSetting(30, Name, classOf[ReqVisFilter]))
```

Comment:
The stats iterator has priority 30.  This iterator should be fired earlier than that so the stats for records with empty visibilities are not analyzed.

Reply:
seems like this could safely go before the default accumulo versioning iterator, no?

Reply:
Assuming that this is settings for scan time only, yes, I think that is reasonable.  

In terms of minc/majc, I don't have a clear rule that'd help determine what to do here.

Reply:
it's applied to all scopes, but really it doesn't seem like it would matter for different scopes - since vis is part of the key, the record would be killed out regardless.